airflow/api_connexion/webserver_auth.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime, timedelta
+from functools import wraps
+from typing import Callable, TypeVar, cast
+
+import jwt
+from flask import Response, current_app, request
+from flask_appbuilder.const import AUTH_LDAP
+from flask_login import login_user
+
+from airflow.configuration import conf
+
+SECRET = conf.get("webserver", "secret_key")
+T = TypeVar("T", bound=Callable)  # pylint: disable=invalid-name
+
+
+def encode_auth_token(user_id):
+    """Generate authentication token"""
+    expire_time = conf.getint("webserver", "session_lifetime_minutes")
+    payload = {
+        'exp': datetime.utcnow() + timedelta(minutes=expire_time),
+        'iat': datetime.utcnow(),
+        'sub': user_id,
+    }
+    return jwt.encode(payload, SECRET, algorithm='HS256')
+
+
+def decode_auth_token(auth_token):
+    """Decode authentication token"""
+    try:
+        payload = jwt.decode(auth_token, SECRET, algorithms=['HS256'])
+        return payload['sub']
+    except (jwt.ExpiredSignatureError, jwt.InvalidTokenError):
+        # Do not raise exception here due to auth_backend auth
+        return None
+
+
+def auth_current_user():
+    """Checks the authentication and return the current user"""
+    auth_header = request.headers.get("Authorization", None)
+    ab_security_manager = current_app.appbuilder.sm
+    token = None
+    user = None
+    if auth_header:
+        try:
+            token = auth_header.split(" ")[1]
+        except IndexError:
+            # Do not raise exception here due to auth_backend auth
+            return None
+    if auth_header and auth_header.startswith("Basic"):
```

Comment:
Something about the way this is integrated still doesn't sit right with me.

This implicitly allows basic auth logins, with no way to disable it, which isn't good.

Also the way this interacts with the _other_ auth backend configured in the app feels a big clunky :(